core/src/main/java/bio/terra/pearl/core/service/workflow/RegistrationService.java

```diff
@@ -39,13 +45,19 @@ public class RegistrationService {
     private EventService eventService;
     private ObjectMapper objectMapper;
 
+    private static final String CHARACTERS = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    private static final String GOVERNED_USER_INDICATOR = "-proxied-";
+    private static final int LENGTH = 10;
```

Comment:
more descriptive name here, and we definitely don't need 10 chars, 4 is plenty given that this is being appended to study/env/proxy specific email
```suggestion
    private static final int GOVERNED_EMAIL_SUFFIX_LENGTH = 4;
```